public/app/plugins/datasource/prometheus/components/PromQueryField.test.tsx

```diff
@@ -140,7 +189,7 @@ describe('PromQueryField', () => {
     let cascader = screen.getByRole('button');
     // Should not show loading
     expect(cascader.textContent).toContain('Metrics');
-    checkMetricsInCascader(cascader, metrics);
+    checkMetricsInCascader(await screen.findByRole('button'), changedMetrics);
```

Comment:
@aocenas correct me if I am wrong, but I think this test should look like this cause we are checking if values changed.

Reply:
not sure exactly, I think the rerender on line 178 should be sync so on line 189 it should already get new DOM. Also the findByRole waits until it finds the element but I think the button should be already there anyway, it does not wait for it to change contents. Was there any issue that this solves?

Reply:
But this test is `'refreshes metrics when time range changes but dont show loading state'` but we've had the same metrics that we were comparing. Not the changed metrics.

Reply:
I think the point is to still show the old metrics. So we don't actually want to wait for any change. Before when you rerendered with range it flashed the `loading` state until you got the new response. This basically checks that right after rerender we still show the old metrics without the loading state.